File: backend/app/util/telefone.py
import re


def limpar_telefone(telefone: str) -> str:
    return re.sub(r"\D", "", telefone or "")


def normalizar_telefone_brasil(telefone: str) -> str:
    limpo = limpar_telefone(telefone)

    if limpo.startswith("55"):
        return limpo

    return f"55{limpo}"


def telefone_basico_valido(telefone: str) -> bool:
    limpo = limpar_telefone(telefone)
    return 10 <= len(limpo) <= 13

File: src/utils/telefone.ts
export function limparTelefone(telefone: string): string {
  return telefone.replace(/\D/g, '');
}

export function telefoneValidoBasico(telefone: string): boolean {
  const limpo = limparTelefone(telefone);
  return limpo.length >= 10 && limpo.length <= 11;
}

export function normalizarTelefoneBrasil(telefone: string): string {
  const limpo = limparTelefone(telefone);

  if (limpo.startsWith('55')) {
    return limpo;
  }

  return `55${limpo}`;
}
